libkbfs/keybase_service_base.go

```diff
@@ -234,12 +239,34 @@ func (k *KeybaseServiceBase) clearCachedUnverifiedKeys(uid keybase1.UID) {
 	delete(k.userCacheUnverifiedKeys, uid)
 }
 
+func (k *KeybaseServiceBase) getCachedTeamInfo(tid keybase1.TeamID) TeamInfo {
+	k.teamCacheLock.RLock()
+	defer k.teamCacheLock.RUnlock()
+	return k.teamCache[tid]
+}
+
+func (k *KeybaseServiceBase) setCachedTeamInfo(
+	tid keybase1.TeamID, info TeamInfo) {
+	k.teamCacheLock.Lock()
+	defer k.teamCacheLock.Unlock()
+	if info.Name == libkb.NormalizedUsername("") {
+		delete(k.teamCache, tid)
+	} else {
+		k.teamCache[tid] = info
+	}
+}
+
 func (k *KeybaseServiceBase) clearCaches() {
 	k.setCachedCurrentSession(SessionInfo{})
-	k.userCacheLock.Lock()
-	defer k.userCacheLock.Unlock()
-	k.userCache = make(map[keybase1.UID]UserInfo)
-	k.userCacheUnverifiedKeys = make(map[keybase1.UID][]keybase1.PublicKey)
+	func() {
+		k.userCacheLock.Lock()
+		defer k.userCacheLock.Unlock()
+		k.userCache = make(map[keybase1.UID]UserInfo)
+		k.userCacheUnverifiedKeys = make(map[keybase1.UID][]keybase1.PublicKey)
+	}()
```

Comment:
As far as I know, the two caches are totally independent and used in different calls, so there shouldn't be any races possible.  If a user call comes races with this function (and hence with the logout), it's undefined whether they will be able to read from the cache or not, and I think that's fine/expected.